### PR TITLE
feat(logo-grid): web component created

### PR DESCRIPTION
**Pull request by RaphaelAmadeu**
### Related Ticket(s)

#3780 

### Description

Web component version created after the react component.

### CODE REVIEW

File: packages/styles/scss/components/logo-grid/_logo-grid.scss [hunk 1]
```diff
@@ -83,7 +84,9 @@
     }
   }
 
-  .#{$prefix}--logo-grid__container {
+  .#{$prefix}--logo-grid__container,
+  :host(#{$dds-prefix}-logo-grid):not(.#{$prefix}--logo-grid__no-border)
```
Comment (asudoh) [inline packages/styles/scss/components/logo-grid/_logo-grid.scss:88]: We don't use CSS classes on our own Web Components, given it tends to clash with CSS classes application code sets:

```suggestion
  :host(#{$dds-prefix}-logo-grid):not([hide-border])
```

File: packages/web-components/examples/codesandbox/components/logo-grid/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "ibmdotcom-web-components-quote-example",
```
Comment (asudoh) [inline packages/web-components/examples/codesandbox/components/logo-grid/package.json:2]:
```suggestion
  "name": "ibmdotcom-web-components-logo-grid-example",
```

File: packages/web-components/src/components/logo-grid/logo-grid.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  static get hideBorderClass() {
+    return `${prefix}--logo-grid__no-border`;
+  }
+
+  @property({ attribute: 'hide-border', reflect: true, type: Boolean })
+  hideBorder = false;
+
+  protected updated() {
+    this.classList.toggle((this.constructor as typeof DDSLogoGrid).hideBorderClass, this.hideBorder);
+  }
```
Comment (asudoh) [inline packages/web-components/src/components/logo-grid/logo-grid.ts:40]: We don't use CSS classes on our own Web Components, given it tends to clash with CSS classes application code sets. We use attribute selectors instead:

```suggestion
```

File: packages/web-components/src/components/logo-grid/logo-grid.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ */
+@customElement(`${ddsPrefix}-logo-grid`)
+class DDSLogoGrid extends StableSelectorMixin(DDSContentBlock) {
+  static get stableSelector() {
+    return `${ddsPrefix}--logo-grid`;
+  }
+
+  static get hideBorderClass() {
+    return `${prefix}--logo-grid__no-border`;
+  }
```
Comment (asudoh) [inline packages/web-components/src/components/logo-grid/logo-grid.ts:33]: We don't use CSS classes on our own Web Components, given it tends to clash with CSS classes application code sets. We use attribute selectors instead:

```suggestion
```

File: packages/web-components/src/components/logo-grid/logo-grid.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  // eslint-disable-next-line class-methods-use-this
+  protected _renderCopy() {
+    return '';
+  }
+
+  // `styles` here is a `CSSResult` generated by custom WebPack loader
+  static get styles() {
+    return css`${super.styles}${styles}`;
+  }
+}
```
Comment (asudoh) [inline packages/web-components/src/components/logo-grid/logo-grid.ts:62]: Nitpick for aligning with other classes in the codebase: Could you please re-group properties/methods, in the following order?

1. Protected methods
2. Public properties
3. Static properties

**Review by asudoh — COMMENTED** [review]
Good to see you keep creating new components @RaphaelAmadeu!

File: packages/web-components/src/components/logo-grid/logo-grid.ts [hunk 6]
```diff
@@ -24,19 +22,11 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
...
-    return `${prefix}--logo-grid__no-border`;
-  }
-
   @property({ attribute: 'hide-border', reflect: true, type: Boolean })
   hideBorder = false;
 
   protected updated() {
-    this.classList.toggle((this.constructor as typeof DDSLogoGrid).hideBorderClass, this.hideBorder);
+    this.toggleAttribute('hide-border', this.hideBorder);
   }
```
Comment (asudoh) [inline packages/web-components/src/components/logo-grid/logo-grid.ts]: `@property({ attribute: 'hide-border', reflect: true, type: Boolean })` takes care of this automatically.


```suggestion
```

**Review by asudoh — COMMENTED** [review]
Thank you for the update @RaphaelAmadeu!

File: packages/web-components/src/components/logo-grid/logo-grid-item.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { customElement, html, property } from 'lit-element';
+// import settings from 'carbon-components/es/globals/js/settings';
```
Comment (asudoh) [inline packages/web-components/src/components/logo-grid/logo-grid-item.ts:10]: Did you mean to remove this line?

File: packages/web-components/src/components/logo-grid/logo-grid-item.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import styles from './logo-grid.scss';
+import DDSImage from '../image/image';
+import StableSelectorMixin from '../../globals/mixins/stable-selector';
+
+// const { prefix } = settings;
```
Comment (asudoh) [inline packages/web-components/src/components/logo-grid/logo-grid-item.ts:16]: Did you mean to remove this line?

File: packages/web-components/src/components/logo-grid/logo-grid.ts [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+ * @element dds-logo-grid
+ */
+@customElement(`${ddsPrefix}-logo-grid`)
+class DDSLogoGrid extends StableSelectorMixin(DDSContentBlock) {
+  @property({ attribute: 'hide-border', reflect: true, type: Boolean })
+  hideBorder = false;
+
+  protected updated() {
+    this.toggleAttribute('hide-border', this.hideBorder);
+  }
```
Comment (asudoh) [inline packages/web-components/src/components/logo-grid/logo-grid.ts:30]: `@property({ attribute: 'hide-border', reflect: true, type: Boolean })` takes care of this automatically.


```suggestion
```

**Review by asudoh — COMMENTED** [review]
Thanks @RaphaelAmadeu for the updates!

File: packages/web-components/src/components/logo-grid/logo-grid.ts (hunk 9 above)
Comment (asudoh) [reply packages/web-components/src/components/logo-grid/logo-grid.ts:30]: @RaphaelAmadeu Have you tried to removing this method? Thanks!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @RaphaelAmadeu for all the updates!

**Comment by jeffchew** [comment]
@RaphaelAmadeu for the documentation, can you provide some more details of what the `complementaryStyleScheme` is, and what other options there are other than `regular`?

**Comment by jeffchew** [comment]
@RaphaelAmadeu can you add stable selectors?

**Pull request merged**